res/controllers/Pioneer-DDJ-SX-scripts.js

```diff
@@ -0,0 +1,2280 @@
+    if (value) {
+        script.toggleControl(group, "headSplit");
+    }
+};
+
+PioneerDDJSX.hotCueButtons = function(channel, control, value, status, group) {
+    var index = control + 1,
+        deck = PioneerDDJSX.channelGroups[group];
+    PioneerDDJSX.cuePressed[deck] = true;
+    script.toggleControl(group, "hotcue_" + index + "_activate");
+};
+
+PioneerDDJSX.clearHotCueButtons = function(channel, control, value, status, group) {
+    var index = control - 0x08 + 1;
+    script.toggleControl(group, "hotcue_" + index + "_clear");
+};
+
+PioneerDDJSX.beatloopButtons = function(channel, control, value, status, group) {
+    var index = control - 0x20,
+        deck = PioneerDDJSX.channelGroups[group];
+    script.toggleControl(
+        group,
+        "beatloop_" + PioneerDDJSX.selectedLoopIntervals[deck][index] + "_toggle"
+    );
+};
+
+PioneerDDJSX.beatloopRollButtons = function(channel, control, value, status, group) {
+    var index = control - 0x10,
+        deck = PioneerDDJSX.channelGroups[group];
+    script.toggleControl(
+        group,
+        "beatlooproll_" + PioneerDDJSX.selectedLooprollIntervals[deck][index] + "_activate"
+    );
+};
+
+PioneerDDJSX.samplerButtons = function(channel, control, value, status, group) {
+    var index = control - 0x30 + 1,
+        deckOffset = PioneerDDJSX.selectedSamplerBank * 8,
+        sampleDeck = "[Sampler" + (index + deckOffset) + "]",
+        playMode = PioneerDDJSX.samplerCueGotoAndPlay ? "cue_gotoandplay" : "start_play";
+
+    if (engine.getValue(sampleDeck, "track_loaded")) {
+        engine.setValue(sampleDeck, playMode, value ? 1 : 0);
+    } else {
+        engine.setValue(sampleDeck, "LoadSelectedTrack", value ? 1 : 0);
+    }
+};
+
+PioneerDDJSX.stopSamplerButtons = function(channel, control, value, status, group) {
+    var index = control - 0x38 + 1,
+        deckOffset = PioneerDDJSX.selectedSamplerBank * 8,
+        sampleDeck = "[Sampler" + (index + deckOffset) + "]",
+        trackLoaded = engine.getValue(sampleDeck, "track_loaded"),
+        playing = engine.getValue(sampleDeck, "play");
+
+    if (trackLoaded && playing) {
+        script.toggleControl(sampleDeck, "stop");
+    } else if (trackLoaded && !playing && value) {
+        script.toggleControl(sampleDeck, "eject");
+    }
+};
+
+PioneerDDJSX.toggleSamplerVelocityMode = function(channel, control, value, status, group) {
+    PioneerDDJSX.samplerVelocityMode = value ? true : false;
+};
+
+PioneerDDJSX.samplerVelocityVolume = function(channel, control, value, status, group) {
+    var index = control - 0x30 + 1,
+        deckOffset = PioneerDDJSX.selectedSamplerBank * 8,
+        sampleDeck = "[Sampler" + (index + deckOffset) + "]",
+        vol = value / 0x7F;
+
+    if (PioneerDDJSX.samplerVelocityMode) {
+        engine.setValue(sampleDeck, "volume", vol);
+    }
+};
+
+PioneerDDJSX.parameterLeft = function(channel, control, value, status, group) {
```

Comment:
Lots of duplication between `parameterLeft` and `parameterRight` functions. May be worth merging them into a single method and passing, or calculating the appropriate bit values to use.